### ISSUE TITLE
Questions about seq2seq model

### ISSUE THREAD

**Issue by Emmyphung** [lorenlugosch/pretrain_speech_model#8]
Hi, thank you for sharing the code and knowledge via the 2 papers! My team and I have been trying to implement your E2E SLU models (both seq2seq and non-seq2seq) and have some questions we would like to ask.

1) Randomness in results 
Even when we run the same models with the same parameters, the results don’t always come out the same, with 1~5% difference. We found the random seed in your code and also in cfg files, but just wondering if different accuracy results are expected.

2) Cross-validation 
We want to know how you did cross-validation for SNIPS dataset. We found 5 fold datasets from the SNIPS so we combined fold_0-3 to be the training data and split fold_4 in half, and set the first half to be validation dataset and the second half to be test dataset. We repeated the same procedure 5 times (set fold_1-4 as training, half of fold0 as valid and another half of fold_0 to be test, ...). We just want to make sure this is the same as how you did.

3) Accuracy calculation 
In the [2nd paper ](https://arxiv.org/abs/1910.09463), you mentioned the “best test accuracy”, and we would like to ask you for clarification on this. Is it possible that it refers to the best validation accuracy achieved during the 40 epochs for each fold, and you average them out over 5 folds? Also, as far as we understand, if a model has possibilities of overfitting, then it’s better to choose more conservative measures instead of the “best” one. 
Excerpt from the paper -- “The model is able to overfit the dataset without synthetic speakers; we therefore record the best test accuracy achieved over the course of training for each fold, instead of the final test accuracy.”

4) Saving model state
I notice that the code in main.py does not save model states based on validation performance but based on epochs. Does this mean that the model run on the test set is the model saved on the last training epoch (even if it’s not the best one)?

**Comment by zhangyanbo2007** [comment]
hello, i also implyment the e2e slu model, but  in seq2seq mode, i face some problems,maybe  we can talk,my whatsapp number is +8615821444815,Looking forward to your reply！
